mmv1/third_party/terraform/website/docs/functions/region_from_id.html.markdown

```diff
@@ -0,0 +1,61 @@
+---
+page_title: region_from_id Function - terraform-provider-google
+description: |-
+  Returns the region within a provided resource id, self link, or OP style resource name.
+---
+
+# Function: region_from_id
+
+Returns the region within a provided resource's id, resource URI, self link, or full resource name.
+
+For more information about using provider-defined functions with Terraform [see the official documentation](https://developer.hashicorp.com/terraform/plugin/framework/functions/concepts).
+
+## Example Usage
+
+### Use with the `google` provider
+
+```terraform
+terraform {
+  required_providers {
+    google = {
+      source = "hashicorp/google"
+    }
+  }
+}
+
+# Value is "us-central1"
+output "function_output" {
+  value = provider::google::region_from_id("https://www.googleapis.com/compute/v1/projects/my-project/regions/us-central1/subnetworks/my-subnetwork")
```

Comment:
Would it be more useful to users to see this used with a resource reference instead/in addition to? Initially, this seems like the more common use case?

Reply:
Yeah that's true. I think any example is going to look a bit odd and a bit contrived (why provide a region when provisioning a regional resource and then derive that value you already have?) but it'd be more useful to a user.

I'll push a commit that [updates resources used in acc tests](https://github.com/GoogleCloudPlatform/magic-modules/pull/10060#discussion_r1520323575) and I'll use the acc test config to update the docs too. Thanks!

Reply:
Go for it! This looks good from my initial review.